packages/webview_flutter/lib/src/webview_method_channel.dart

```diff
@@ -46,6 +46,12 @@ class MethodChannelWebViewPlatform implements WebViewPlatformController {
       case 'onPageStarted':
         _platformCallbacksHandler.onPageStarted(call.arguments['url']!);
         return null;
+      case 'onPullToRefresh':
+        final Future<void> refreshResult =
+            _platformCallbacksHandler.onPullToRefresh();
+        await refreshResult;
```

Comment:
If this is a user provided callback, it might throw and left the platform-side in an incorrect state.
You might want to use:
```dart
try {
  // call user callback.
} finally {
  await _channel.invokeMethod('endRefreshing', null);
}
```